hs/src/Reach/Verify/SMT.hs

```diff
@@ -711,13 +712,20 @@ pathAddUnbound at_dv (Just dv) msmte = do
   let smlet = Just . SMTLet at_dv dv (DLV_Let DVC_Once dv) Witness =<< msmte
   pathAddUnbound_v v t smlet
 
+assertInvariants :: SrcLoc -> DLType -> String -> App ()
+assertInvariants at_dv t v =
+  when (t == T_UInt) $ do
+        rhs <- smt_c at_dv DLC_UInt_max
+        smtAssert (smtApply "<=" [Atom v, rhs])
+
 pathAddBound :: SrcLoc -> Maybe DLVar -> Maybe SMTExpr -> SExpr -> SMTCat -> App ()
 pathAddBound _ Nothing _ _ _ = mempty
 pathAddBound at_dv (Just dv) de se sc = do
   let DLVar _ _ t _ = dv
   v <- smtVar dv
   let smlet = Just . SMTLet at_dv dv (DLV_Let DVC_Once dv) sc =<< de
   smtDeclare_v v t smlet
+  assertInvariants at_dv t v
```

Comment:
@jeapostrophe  Any harm in this? Otherwise, making claims about balances in loop invariants require `balance(token) <= UInt.max`

Reply:
No harm